### PR TITLE
Refine stubs for Python 2's decimal module

### PR DESCRIPTION
**Pull request by Michael0x2a**
The decimal module for Python 2 was relatively incomplete, unlike the decimal module for Python 3. This commit copies the relevant type signatures from Python 3's decimal module to Python 2's (and slightly tweaks them to match the Python 2 function definitions in a few places).

There was a lot of code in both stubs and it wasn't clear to me if it was safe to merge the two modules together, so I refrained from doing so.

### CODE REVIEW

File: stdlib/2.7/decimal.pyi [hunk 1]
```diff
@@ -1,17 +1,25 @@
...
 
-ROUND_DOWN = ...  # type: Any
-ROUND_HALF_UP = ...  # type: Any
-ROUND_HALF_EVEN = ...  # type: Any
-ROUND_CEILING = ...  # type: Any
-ROUND_FLOOR = ...  # type: Any
-ROUND_UP = ...  # type: Any
-ROUND_HALF_DOWN = ...  # type: Any
-ROUND_05UP = ...  # type: Any
+_Decimal = Union[Decimal, int]
```
Comment (gvanrossum) [inline stdlib/2.7/decimal.pyi]: I think modern Decimal is lenient about floats and just accepts them. I recommend just adding float here, else you have to add it to many other places (in our server code I found problems comparing floats to decimals.

File: stdlib/2.7/decimal.pyi [hunk 2]
```diff
@@ -40,120 +48,125 @@ class Overflow(Inexact, Rounded): ...
...
-def setcontext(context): ...
-def getcontext(): ...
-def localcontext(ctx=None): ...
+def setcontext(context: Context): ...
+def getcontext() -> Context: ...
+def localcontext(ctx: Optional[Context] = None) -> _ContextManager: ...
 
 class Decimal(SupportsAbs[Decimal], SupportsFloat, SupportsInt):
-    def __new__(cls, value=..., context=None): ...
+    def __init__(cls, value: Union[_Decimal, float, str,
```
Comment (gvanrossum) [inline stdlib/2.7/decimal.pyi]: Can you also allow Number here?

File: stdlib/2.7/decimal.pyi (hunk 1 above)
Comment (Michael0x2a) [inline stdlib/2.7/decimal.pyi]: Sounds good. Should I also make that change in the Python 3's version of Decimal? (I copied `_Decimal` from the current Python 3 stubs).

Comment (gvanrossum) [inline stdlib/2.7/decimal.pyi]: Hm, this needs more nuance. IIUC comparisons with floats are acceptable, but not e.g. addition. So maybe you need a more lenient union type just for comparisons.

File: stdlib/2.7/decimal.pyi (hunk 2 above)
Comment (gvanrossum) [inline stdlib/2.7/decimal.pyi]: NM, I think this a problem on our side.

**Comment by Michael0x2a** [comment]
@gvanrossum -- ok, I made comparisons more lenient. Try now?

**Comment by gvanrossum** [comment]
I think this is a good improvement and I think it's found some nonsense on our side!

**Pull request merged**